### ISSUE TITLE
Different ports for load balancer and docker container

### ISSUE THREAD

**Issue by Ilchuk-Mihail** [pulumi/examples#267]
I use simple containerization deployment : 

```
const pulumi = require("@pulumi/pulumi");
const cloud = require("@pulumi/cloud-aws");

let service = new cloud.Service("pulumi-nginx", {
    containers: {
        nginx: {
            build: "./app",
            memory: 512,
            ports: [{ port: 80 }],
        },
    }
});

// export just the hostname property of the container frontend
exports.hostname = pulumi.interpolate `http://${service.defaultEndpoint.hostname}`;

```
My container EXPOSE port 3000. 
But in result, I have load balancer which have listener to port 80  and forwarded this query (via private ip) to container port - also 80 . (and it doesn't work) 

If I change ` ports: [{ port: 3000}] `  - I have listener 3000 -> forwarded to container port 3000 . 
And it works when I go to url with specified port -  http://.........elb.us-east-1.amazonaws.com:3000
and doesn't work without port. 

How to do that: listener points 80 port -> and forward to specified container port 3000 ?
And the same for 443 (ssl) :)

So, in a nutshell - Different port for listener and container.

**Comment by lukehoban** [comment]
There is a `targetPort` property you can set to target port `3000` on the container.  See https://pulumi.io/reference/pkg/nodejs/@pulumi/cloud/#ContainerPort-targetPort.

**Comment by cloudpaper** [comment]
Leaving this here for anyways who might come across it later.

This does *not* work for Fargate services. If you're running on Fargate you have to use the same port as what your container is listening on.

**Comment by xai1983kbu** [comment]
hello @cloudpaper
What is your solution?

Can we Dockerize next.js and nginx at one Dockerfile (https://stackoverflow.com/a/63038894/9783262)?

**Comment by xai1983kbu** [comment]
I found working solution without Nginx
https://stackoverflow.com/a/61134810/9783262
It's interesting now - Is it the right way (don't use Nginx)?
